src/__tests__/container.test.ts

```diff
@@ -174,6 +174,66 @@ describe('container', () => {
       expect(err.message).toContain('first -> second -> third')
     })
 
+    it('allows setting defaultArgs on the class', () => {
+      const container = createContainer()
+      class First {
+        constructor(cradle: any) {
+          this.unregistered = cradle.unregistered
+        }
+        unregistered: any
+        static [RESOLVER] = {
+          lifetime: Lifetime.SCOPED,
+          defaultArgs: {
+            unregistered: 'bar',
+          },
+        }
+      }
+      container.register({
+        first: asClass(First),
+      })
+
+      const first = container.resolve('first')
+      expect(first.unregistered).toEqual('bar')
+    })
+
+    it('allows setting defaultArgs at registration', () => {
+      const container = createContainer()
+      class First {
+        constructor(cradle: any) {
+          this.unregistered = cradle.unregistered
+        }
+        unregistered: any
+      }
+      container.register({
+        first: asClass(First).withDefaultArgs({ unregistered: 'bar' }),
+      })
+
+      const first = container.resolve('first')
+      expect(first.unregistered).toEqual('bar')
+    })
+
+    it('allows setting defaultArgs in the constructor', () => {
+      const container = createContainer()
+      class First {
+        constructor({ unregistered = 'bar' }) {
+          this.unregistered = unregistered
+        }
+        unregistered: any
+        static [RESOLVER] = {
+          lifetime: Lifetime.SCOPED,
+          defaultArgs: {
+            unregistered: 'bar',
```

Comment:
Shouldn't this be `unregistered: undefined` to test the constructor argument is used?

Reply:
Ah yeah, I changed in the `withDefaultArgs` call but not here